src/Pool.sol

```diff
@@ -688,81 +451,7 @@ abstract contract Pool {
-        address receiverAddr = ReceiverWeightsImpl.ADDR_ROOT;
-        address receiverHint = ReceiverWeightsImpl.ADDR_ROOT;
-        while (true) {
-            uint32 weight;
-            (receiverAddr, receiverHint, weight, ) = proxy.receiverWeights.nextWeightPruning(
-                receiverAddr,
-                receiverHint
-            );
-            if (receiverAddr == ReceiverWeightsImpl.ADDR_ROOT) break;
-            require(receiversCount < PROXY_WEIGHTS_COUNT_MAX, "Too many proxy receivers");
-            receiversList[receiversCount++] = ReceiverWeight(receiverAddr, weight);
-        }
-        // The proxy doesn't exist
-        if (receiversCount == 0) return;
-
-        // Iterating over deltas, see `ProxyDeltas` for details
-        uint64 cycle = ProxyDeltasImpl.CYCLE_ROOT;
-        uint64 cycleHint = ProxyDeltasImpl.CYCLE_ROOT;
-        uint64 finishedCycle = currTimestamp() / cycleSecs;
-        uint64 currCycle = finishedCycle + 1;
-        // The sum of all the future changes to the per-second amount the proxy receives.
-        // This is also the per-second amount the proxy receives per weight in the current cycle,
-        // but with its sign inverted.
-        // Thus if `multiplier` is `1`, then this value is negative and if `-1`, it's positive.
-        int128 totalDelta = 0;
-        while (true) {
-            int128 thisCycleDelta;
-            int128 nextCycleDelta;
-            (cycle, cycleHint, thisCycleDelta, nextCycleDelta) = proxy
-                .amtPerWeightDeltas
-                .nextDeltaPruning(cycle, cycleHint, finishedCycle);
-            if (cycle == ProxyDeltasImpl.CYCLE_ROOT) break;
-            // `thisCycleDelta` from the previously finished cycle is irrelevant
-            if (cycle == finishedCycle) thisCycleDelta = 0;
-            thisCycleDelta *= multiplier;
-            nextCycleDelta *= multiplier;
-            totalDelta += thisCycleDelta + nextCycleDelta;
-            for (uint32 i = 0; i < receiversCount; i++) {
-                Receiver storage receiver = receivers[receiversList[i].receiver];
-                uint32 weight = receiversList[i].weight;
-                receiver.amtDeltas[cycle].thisCycle += int128(uint128(weight)) * thisCycleDelta;
-                receiver.amtDeltas[cycle].nextCycle += int128(uint128(weight)) * nextCycleDelta;
-            }
-        }
-        // Set the delta for the current cycle, which balances all the applied deltas
-        if (totalDelta != 0) {
-            for (uint32 i = 0; i < receiversCount; i++) {
-                Receiver storage receiver = receivers[receiversList[i].receiver];
-                int128 weight = int128(uint128(receiversList[i].weight));
-                receiver.amtDeltas[currCycle].thisCycle -= totalDelta * weight;
-            }
-        }
-    }
-
-    function currTimestamp() public view returns (uint64) {
+    function _currTimestamp() internal view returns (uint64) {
```

Comment:
Why not public if it is a `view`?

Reply:
Because it's not a useful public API member. Who want to use it? It's also much harder to make any changes in a public API.